app/products/calls/components/incoming_calls_container.tsx

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {StyleSheet, View} from 'react-native';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+
+import CallNotification from '@calls/components/call_notification';
+import {useCurrentCall, useGlobalCallsState, useIncomingCalls} from '@calls/state';
+import {
+    CALL_ERROR_BAR_HEIGHT,
+    CURRENT_CALL_BAR_HEIGHT,
+    DEFAULT_HEADER_HEIGHT,
+    JOIN_CALL_BAR_HEIGHT,
+} from '@constants/view';
+
+const topBarHeight = DEFAULT_HEADER_HEIGHT;
+
+const style = StyleSheet.create({
+    wrapper: {
+        position: 'absolute',
+        width: '100%',
+        marginTop: 8,
+        gap: 8,
+    },
+});
+
+type Props = {
+    channelId: string;
+    showingJoinCallBanner: boolean;
+    showingCurrentCallBanner: boolean;
+    threadScreen?: boolean;
+}
+
+export const IncomingCallsContainer = ({
+    channelId,
+    showingJoinCallBanner,
+    showingCurrentCallBanner,
+    threadScreen,
+}: Props) => {
+    const incomingCalls = useIncomingCalls().incomingCalls;
+    const insets = useSafeAreaInsets();
+    const micPermissionsGranted = useGlobalCallsState().micPermissionsGranted;
+    const currentCall = useCurrentCall();
+
+    // If we're in the channel for the incoming call, don't show the incoming call banner.
+    const calls = incomingCalls.filter((ic) => ic.channelID !== channelId);
+    if (calls.length === 0) {
+        return null;
+    }
```

Comment:
I assume the ringing feature being enabled or not is implicitly checked here?

Reply:
It's checked earlier, in the `processIncomingCalls` functions. If it's not enabled we'll never get to this point (there would be no incoming calls in the state).